GoogleChromeLabs/ps-analysis-tool
# Feature: Add Cookie Table and Details Components with Tabular UI

## Description


This pull request introduces: 
- Custom `Table` module based on `react-table` to facilitate dynamic and interactive table functionalities for the cookie management system.
- `CookieTable` component, which serves as a layer on top of the custom `Table` module. This component provides a comprehensive listing of all cookies and their attributes in a structured table format.
- `CookieDetails` component to provide users with a detailed preview of the selected cookie's value and additional information.

## Relevant Technical Choices
- `@tanstack/react-table` Library: We opted to use the react-table library to build the custom Table module due to its extensive feature set and ease of use. It provides good data-handling, resizing functionalities and ability to customise UI as per our requirements.
- The `Table` component have been created into smaller, more manageable code segments, enhancing modularity and making them easily usable as subsets in any component that requires it's functionality.
- The `CookieTable` and `CookieDetails` components as separate entities, we ensure a modular and maintainable codebase.


## Testing Instructions
- Open the terminal in the root directory of the project.
- Run the command `npm run dev`, and the extension' Webpack development will spin up.
- Open website `indianexpress.com`, open devtools panel and select `Privacy Sandbox` tab.
- Under the `Privacy Sandbox` tab, you'll be able to see Cookie Table and Details component.
- Try to resize the table columns by selecting right side of their header.
- Try to resize the Cookie Details component by holding the upper border of component.


## Additional Information:



## Screenshot/Screencast


https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/58820001/0b4f025b-bd26-4be0-81b9-9a63f58af3d1





---



Fixes #68
